calcom/cal.com
# Stripe add the ability to place hold on cards

## What does this PR do?



This PR allows the Stripe app to place holds on cards and allow users to charge a no-show fee to attendees.

<img width="884" alt="image" src="https://user-images.githubusercontent.com/65426560/229940238-29b2dc2d-9d70-4ef3-a209-20850b8930c7.png">

<img width="595" alt="image" src="https://user-images.githubusercontent.com/65426560/229943763-24a2dcbc-0f67-4237-b72e-87940cc136ee.png">

<img width="1152" alt="image" src="https://user-images.githubusercontent.com/65426560/229943984-9a0eed85-3b33-444e-a742-4152951be48d.png">

<img width="583" alt="image" src="https://user-images.githubusercontent.com/65426560/229944063-2a04c71d-caa6-4a64-9d54-19795c012c5d.png">


Fixes # (issue)



**Environment**: Staging(main branch) / Production

## Type of change



- New feature (non-breaking change which adds functionality)

## How should this be tested?



- Ensure that you are forwarding Stripe webhooks locally
- Create an event type and add a price with the option to charge a no-show fee
- Create a booking and enter the Stripe testing card
    - Enter an invalid expiry date to render the error message on the payment page
- As the organizer go to the booking page
    - Charge the no-show fee to the attendee
    - There should be an email sent to the attendee
- Create a booking, but this time with a Stripe declined test card
- If you try to charge the no-show fee, an error message should appear

## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
